inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_b_fs_zyx_fsv16_imad.cl

```diff
@@ -44,6 +55,231 @@ inline ACCUMULATOR_TYPE FUNC(apply_pooling)(ACCUMULATOR_TYPE tmp, ACCUMULATOR_TY
 #endif
 }
 
+#if GLOBAL_POOLING
+__attribute__((intel_reqd_sub_group_size(FEATURE_SLICE_SIZE)))
+__attribute__((reqd_work_group_size(1, LWS, 1)))
+KERNEL(pooling_gpu_b_fs_zyx_fsv16)(
+    const __global INPUT0_TYPE* input,
+    __global OUTPUT_TYPE* output
+#if HAS_FUSED_OPS_DECLS
+    , FUSED_OPS_DECLS
+#endif
+) {
+    const uint x       = (uint)get_global_id(0);
+    const uint set_idx = (uint)get_global_id(1);
+#if OUTPUT_DIMS == 4
+    const uint y    = set_idx;
+    const uint z    = 0;
+#else
+    const uint y    = set_idx % INPUT0_SIZE_Y;
+    const uint z    = set_idx / INPUT0_SIZE_Y;
+#endif
+    const uint bf   = (uint)get_global_id(2);
+    const uint f    = (bf * FEATURE_SLICE_SIZE) % ALIGN_TO(INPUT0_FEATURE_NUM, FEATURE_SLICE_SIZE);
+    const uint b    = (bf * FEATURE_SLICE_SIZE) / ALIGN_TO(INPUT0_FEATURE_NUM, FEATURE_SLICE_SIZE);
+
+    const uint loc_id   = (uint)get_local_id(1);
+
+    const bool last_in_f_group = (f == FEATURE_SLICE_SIZE * ((INPUT0_FEATURE_NUM - 1) / FEATURE_SLICE_SIZE));
```

Comment:
As I can see, you use this variable only under this macro:
`#if INPUT0_FEATURE_NUM % FEATURE_SLICE_SIZE != 0`
So it might make sense to add this macro here